rest_framework/utils/field_mapping.py

```diff
@@ -266,6 +266,9 @@ def get_relation_kwargs(field_name, relation_info):
             # If this field is read-only, then return early.
             # No further keyword arguments are valid.
             return kwargs
+        limit_choices_to = model_field.get_limit_choices_to()
+        if limit_choices_to:
+            kwargs['queryset'] = kwargs['queryset'].filter(**limit_choices_to)
```

Comment:
I think this block should probably go above the `if has_through_model:` switch. Since there's a couple of cases where the `queryset` keyword argument gets popped off.

Reply:
Or else use `if queryset in kwargs and limit_choices_to:`, which would also do the trick.

Reply:
You're right. I did not think it through. I just wanted to avoid extra work putting it after the read_only but forgot that `queryset` can be popped.

Maybe another solution would be to add a check like `'queryset' in kwargs`?

Reply:
I was too slow to post my comment and did not see yours. :/